lib/prompts.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+var _ = require("lodash");
+var async = require("async");
+var inquirer = require("inquirer");
+
+/**
+ * Prompt user for input, validate, and add derived fields to final data object.
+ *
+ * @param {Object}    init      Initialization configuration (`prompts`, `derived`)
+ * @param {Function}  callback  Calls back with `(err, data)`
+ * @returns {void}
+ */
+module.exports = function (init, callback) {
+  if (!init) { return callback(new Error("Invalid init object")); }
+
+  // Mutate `prompts` into appropriate format.
+  var prompts = init.prompts || [];
+  if (!_.isArray(prompts)) {
+    if (!_.isObject(prompts)) {
```

Comment:
I know it would mean doing the `_.isArray` check twice but it might be better to move this into a short circuit evaluation up at the top with the other error condition.

This is the whole lodash _.isArray fn so I'd say in terms of perf you're pretty safe to call it twice in the name of code clarity.
```js
    var isArray = nativeIsArray || function(value) {
      return isObjectLike(value) && isLength(value.length) && objToString.call(value) == arrayTag;
    };
```